components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1230,101 +1233,71 @@ subroutine activate_modal(wbar, sigw, wdiab, wminf, wmaxf, tair, rhoair,  &
-   real(r8) dqsdt ! change in qs with temperature
-   real(r8) dqsdp ! change in qs with pressure
-   real(r8) g ! thermodynamic function (m2/s)
-   real(r8) zeta(nmode), eta(nmode)
-   real(r8) lnsmax ! ln(smax)
-   real(r8) alpha
-   real(r8) gamma
-   real(r8) beta
-   real(r8) sqrtg(nmode)
-   real(r8) :: amcube(nmode) ! cube of dry mode radius (m)
-   real(r8) :: smcrit(nmode) ! critical supersatuation for activation
-   real(r8) :: lnsm(nmode) ! ln(smcrit)
-   real(r8) smc(nmode) ! critical supersaturation for number mode radius
-   real(r8) sumflx_fullact
-   real(r8) sumflxn(nmode)
-   real(r8) sumflxm(nmode)
-   real(r8) sumfn(nmode)
-   real(r8) sumfm(nmode)
-   real(r8) fnold(nmode)   ! number fraction activated
-   real(r8) fmold(nmode)   ! mass fraction activated
-   real(r8) wold,gold
-   real(r8) alogam
-   real(r8) rlo,rhi,xint1,xint2,xint3,xint4
-   real(r8) wmin,wmax,w,dw,dwmax,dwmin,wnuc,dwnew,wb
-   real(r8) dfmin,dfmax,fnew,fold,fnmin,fnbar,fsbar,fmbar
-   real(r8) alw,sqrtalw
-   real(r8) smax
-   real(r8) x,arg
-   real(r8) xmincoeff,xcut,volcut,surfcut
-   real(r8) z,z1,z2,wf1,wf2,zf1,zf2,gf1,gf2,gf
-   real(r8) etafactor1,etafactor2(nmode),etafactor2max
-   integer m,n
-   !      numerical integration parameters
-   real(r8), parameter :: eps=0.3_r8,fmax=0.99_r8,sds=3._r8
-
-   real(r8), parameter :: namin=1.e6_r8   ! minimum aerosol number concentration (/m3)
-
-   integer ndist(nx)  ! accumulates frequency distribution of integration bins required
-   data ndist/nx*0/
-   save ndist
+   real(r8), parameter :: p0 = 1013.25e2_r8    ! reference pressure [Pa]
+   real(r8) pres ! pressure [Pa]
+   real(r8) diff0 ! vapor diffusivity [m2/s]
+   real(r8) conduct0 ! thermal conductivity [J / (m-s-K)]
+   real(r8) es ! saturation vapor pressure [Pa]
+   real(r8) qs ! water vapor saturation specific humidity [kg/kg]
+   real(r8) dqsdt ! change in qs with temperature  [(kg/kg)/T]
+   real(r8) zeta(nmode), eta(nmode) ! [dimensionless]
+   real(r8) alpha ! [/m]
+   real(r8) gamma ![m3/kg]
+   real(r8) beta ! [m2/s]
+   real(r8) gthermfac ! thermodynamic function [m2/s]
+   real(r8) :: amcube(nmode) ! cube of dry mode radius [m3]
+   real(r8) smc(nmode) ! critical supersaturation for number mode radius [fraction]
+   real(r8) :: lnsm(nmode) ! ln(critical supersaturation for activation) [dimensionless]
+
+   real(r8) wnuc  ! nucleation w, but = wbar if wdiab == 0 [m/s]
+   real(r8) alw ! [/s]
+   real(r8) smax ! maximum supersaturation [fraction]
+   real(r8) lnsmax ! ln(smax) [dimensionless]
+   real(r8) arg_erf_n,arg_erf_m  ! [dimensionless]
+   real(r8) etafactor1  ! [/ s^(3/2)]
+   real(r8) etafactor2(nmode),etafactor2max ! [s^(3/2)]
+   integer imode
+
 
    fn(:)=0._r8
    fm(:)=0._r8
    fluxn(:)=0._r8
    fluxm(:)=0._r8
    flux_fullact=0._r8
 
-   if(nmode.eq.1.and.na(1).lt.1.e-20_r8)return
 
-   if(sigw.le.1.e-5_r8.and.wbar.le.0._r8)return
+  if(nmode == 1 .and. na(1) < 1.e-20_r8)return
```

Comment:
I am not sure if `nmode` can be 1 for any case. `nmode` in our case is always 4, so we can remove this statement.